docs/dev-architecture.md

```diff
@@ -0,0 +1,90 @@
+# Developer Architecture Overview
+
+This shall give a sufficiently good overview over the haskell.nix
+ideas, such that a new developer can navigate around without too
+much trouble.
+
+## Packages
+
+`haskell.nix` is centered around packages (haskell package
+descriptions as nix-expressions). These are generated by
+`cabal-to-nix` from the `nix-tools` package. `stack-to-nix` and
+`plan-to-nix` will delegate the transformation of cabal packages to
+nix expressions to the same code that `cabal-to-nix` uses.
+
+These packages will look similar to the following:
+
+```nix
+{ system, compiler, flags, pkgs, hsPkgs, pkgconfPkgs, ... }:
+{
+  flags = {};
+  package = { ... };
+  components = {
+    "library" = { depends = [ ... ]; };
+	exes = { "..." = { depends = [ ... ]; }; ... };
+	sublibs = { "..." = { depends = [ ... ]; }; ... };
+	tests = { "..." = { depends = [ ... ]; }; ... };
+	benchmarks = { "..." = { depends = [ ... ]; }; ... };
```

Comment:
Looks like something happened with the indentation.

Reply:
thanks. stupid tabs.